src/actions/actions.ts

```diff
@@ -3673,6 +3673,16 @@ abstract class MoveByScreenLine extends BaseMovement {
   value: number = 1;
 
   public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    // Because we can't support moving by screen line when in visualLine mode,
+    // we change to moving by regular line in visualLine mode. We can't move by
+    // screen line is that our ranges only support a start and stop attribute,
+    // and moving by screen line just snaps us back to the original position.
+    // Check PR #1600 for discussion.
+    if (vimState.currentMode === ModeName.VisualLine) {
+      if (this.movementType === "up" || this.movementType === "down") {
```

Comment:
Do you think it's better to define another class "MoveUpByScreenLineVisualLine"?